src/common/transformations/tests/op_conversions/sdpa_to_paged_attention_test.cpp

```diff
@@ -0,0 +1,537 @@
+//
+
+#include "openvino/pass/sdpa_to_paged_attention.hpp"
+
+#include <gtest/gtest.h>
+
+#include "common_test_utils/ov_test_utils.hpp"
+#include "openvino/core/model.hpp"
+#include "openvino/op/add.hpp"
+#include "openvino/op/broadcast.hpp"
+#include "openvino/op/concat.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/op/divide.hpp"
+#include "openvino/op/gather.hpp"
+#include "openvino/op/matmul.hpp"
+#include "openvino/op/multiply.hpp"
+#include "openvino/op/ops.hpp"
+#include "openvino/op/paged_attention.hpp"
+#include "openvino/op/power.hpp"
+#include "openvino/op/reduce_mean.hpp"
+#include "openvino/op/reshape.hpp"
+#include "openvino/op/scaled_dot_product_attention.hpp"
+#include "openvino/op/select.hpp"
+#include "openvino/op/shape_of.hpp"
+#include "openvino/op/sqrt.hpp"
+#include "openvino/op/squeeze.hpp"
+#include "openvino/op/subtract.hpp"
+#include "openvino/op/transpose.hpp"
+#include "openvino/op/unsqueeze.hpp"
+#include "transformations/sdpa_to_paged_attention/total_sequence_length_pattern.hpp"
+#include "transformations/utils/gen_pattern.hpp"
+#include "transformations/utils/print_model.hpp"
+
+using namespace ov;
+using namespace std;
+using namespace testing;
+using namespace ov::op;
+using namespace ov::gen_pattern;
+
+namespace {
+
+// Constants and Parameters attributes:
+auto el_type_i64 = std::pair<std::string, detail::AttrAny>({"element_type", "i64"});
+auto el_type_i32 = std::pair<std::string, detail::AttrAny>({"element_type", "i32"});
+auto el_type_f32 = std::pair<std::string, detail::AttrAny>({"element_type", "f32"});
+
+// Convert ops attributes:
+auto dest_type_i64 = std::pair<std::string, detail::AttrAny>({"destination_type", "i64"});
+auto dest_type_f32 = std::pair<std::string, detail::AttrAny>({"destination_type", "f32"});
+auto dest_type_f16 = std::pair<std::string, detail::AttrAny>({"destination_type", "f16"});
+
+// Other attributes:
+auto numpy_broadcast = std::pair<std::string, detail::AttrAny>({"auto_broadcast", "numpy"});
+auto special_zero_true = std::pair<std::string, detail::AttrAny>({"special_zero", true});
+
+auto single_val = [](int rank, float val) {
+    return makeConst(element::f32, ov::Shape{std::vector<size_t>(rank, 1)}, {val});
+};
+
+ov::ParameterVector nodes_to_params(const ov::NodeVector& node_vec) {
+    ov::ParameterVector params;
+    params.reserve(node_vec.size());
+    for (const auto& node : node_vec) {
+        params.push_back(ov::as_type_ptr<v0::Parameter>(node));
+    }
+    return params;
+}
+
+enum QKV : int { Q = 0, K = 1, V = 2 };
+vector<int> MOCK_VALUE = {1};
+
+class Qwen7bChatSDPA {
+public:
+    static std::shared_ptr<Node> gen_embeddings(const std::shared_ptr<Node>& input_ids) {
+        auto view_reshape = makeOP<v1::Reshape>({input_ids, {-1, 0}}, {special_zero_true});
+        auto input_ids_i64 = makeOP<v0::Convert>({view_reshape}, {dest_type_i64});
+
+        auto weights = makeConst(element::u8, {151936, 4096}, MOCK_VALUE);
```

Comment:
Why do we need to have such a big size in a test? It is 600MB.
How is this part of the model graph related to proposed changes?

Reply:
Good catch!
I forgot to cut the weights :(

Reply:
done